client/components/domains/register-domain-step/index.jsx

```diff
@@ -299,6 +301,16 @@ const RegisterDomainStep = React.createClass( {
 		} );
 	},
 
+	getSuggestionToken: function() {
+		const store = pick( SignupDependencyStore.get(), 'designType' );
```

Comment:
This should definitely be a prop - RegisterDomainStep is shared between signup and domain management and should not depend on the `SignupDependencyStore` directly. It's used by context-specific components - https://github.com/Automattic/wp-calypso/blob/a70587157b33783ba4d5e0e4a1e136680d459d51/client/signup/steps/domains/index.jsx#L179-L196 (signup) and https://github.com/Automattic/wp-calypso/blob/3d316ea3577219a867f0858210a9cb10a084a871/client/my-sites/domains/domain-search/domain-search.jsx#L123-L134 (domain management).
You can see that there's already a bit of logic there, depending on the context/flow. `suggestionToken` should be a property on this component and can have a default value of `default` (via `defaultProps`), while the signup parent would pass one from the `SignupDependencyStore`.